pkg/crypto/tinkcrypto/primitive/composite/ecdh1pu/ecdh1pu_decrypt_factory.go

```diff
@@ -62,30 +62,47 @@ func newDecryptPrimitiveSet(ps *primitiveset.PrimitiveSet) (*decryptPrimitiveSet
 	return ret, nil
 }
 
+var errNotApplicable = errors.New("not applicable")
+
+func (a *decryptPrimitiveSet) tryNonRawKeys(ct, aad []byte) ([]byte, error) {
```

Comment:
@Baha-sk fyi

Reply:
@troyronda @Baha-sk the function was modified due to `nestif`. I noticed that we have two same functions.

Reply:
Tink actually changed the for loop to start from the end of the entries array (latest entry representing most recent key in the keyset), I was going to modify this code as part of Tink 1.4 upgrade after legacyKMS removal change that I'm currently working on. I'll add my change on top of yours @soluchok.

my recommendation: instead of computing ctNoPrefix and entries in this function, move them out and pass them as the `ct` argument and add a new `entries` argument in this new function. This way, you can also replace the original lines 93 to 105 further down to call this new function (by passing the original `ct` and `a.ps.RawEntries()` as `entries` argument)

Reply:
@Baha-sk  Fixed. Pls, check it. I did not modify existing logic. All tests are the same.

Reply:
@soluchok it's not exactly what I meant, but it's ok, now you call Decrypt() only once. It seems OK since It avoids calling Decrypt twice. I still need to update the code later on to inverse the iterations in the for loop.